tests/src/nodes/nomos.rs

```diff
@@ -182,15 +171,17 @@ impl Node for NomosNode {
                 // all nodes that will be subsequently spawned.
                 // If not, the leader will miss votes from nodes spawned before itself.
                 // This issue will be resolved by devising the block catch-up mechanism in the future
-                let mut nodes = vec![Self::spawn(configs.swap_remove(next_leader_idx)).await];
-                let listening_addr = nodes[0].get_listening_address().await;
-                for mut conf in configs {
-                    conf.network
-                        .backend
-                        .initial_peers
-                        .push(listening_addr.clone());
+                configs.swap(0, next_leader_idx);
+
+                let mut nodes = Vec::new();
+                for (i, conf) in configs.clone().iter_mut().enumerate() {
+                    if i > 0 {
+                        let prev_node_port = configs[i - 1].network.backend.inner.port;
+                        let addr = multiaddr!(Ip4([127, 0, 0, 1]), Tcp(prev_node_port));
+                        conf.network.backend.initial_peers.push(addr);
```

Comment:
Just so I understand, we make a chain now?

Reply:
Yes, right. But, I'll try the way that @zeegomo suggested below: https://github.com/logos-co/nomos-node/pull/442#pullrequestreview-1652771120